plonk/src/circuit/customized/ecc/glv.rs

```diff
@@ -521,7 +514,8 @@ where
     };
 
     //  (f.3) either f.1 or f.2 is satisfied
-    let sat = circuit.conditional_select(k2_sign_var, k2_is_neg_sat, k2_is_pos_sat)?;
+    let sat =
+        circuit.conditional_select(k2_sign_var, k2_is_neg_sat.into(), k2_is_pos_sat.into())?;
```

Comment:
looks like `k2_is_pos_sat` is an output of `circuit.check_equal(left_var, right_var)?` which makes me think that `circuit.check_equal(left_var, right_var)?` should output a BoolVar instead of a `Variable`. WDYT?

Reply:
yeah exactly, currently we have: `pub fn check_equal(&mut self, a: Variable, b: Variable) -> Result<BoolVar, PlonkError> ` which is why you see here `k2_is_pos_sat.into()` since `k2_is_pos_sat` is `BoolVar` type.

Reply:
cool. Sorry I somehow thought the `into()` here was the reversed direction